### ISSUE TITLE
Add a healthcheck endpoint to all services

### ISSUE THREAD

**Issue by jbonnet** [sonata-nfv/son-gkeeper#370]
We should add health (or liveliness) checks endpoint. this could be a `HEAD` or a `GET`. The later can return (e.g.) the time the service is up, though with the use of `Kubernetes` this is not as relevant (since a service can be restarted automatically).

For the **API** (`son-gtkapi`) it might make sense to distinguish between the health of the **API** itself and the health of the other microservices.

An **alternative** to this approach would be simply to **log** regularly, or publish a **KPI**. These records could be followed externally.

**Comment by felipevicens** [comment]
@jbonnet First, the simplest way is add a raw html page with the response ":-)" very useful for service up check as well load balancing and platform updates. We could add this in each api route as /api/healthcheck

We can go further and add other kind of verifications before the :-) like when the /api/healthcheck is requested, then connect to databases, read files, check connections. I think `son-gtkapi` should check each health check of each component downstream and report any failure.

What do you think?

**Comment by jbonnet** [comment]
@felipevicens Thanks for the suggestions.

For the HTML, we could have a **dashboard** (in the **GUI**? See [Status.io's status page, e.g.](https://status.status.io/) ). But one thing is the view (your suggestion, in HTML, for humans) and another is a tool friendly way of being checked. And also the process that implements the collection of data.

**Comment by felipevicens** [comment]
Yes exactly, 

**humans** = html data ":-)" == healthy | html data "error XXX" == unhealthy
**machines** = http code "200 OK" == healthy | http code "4XX/5XX"  == unhealthy

Same protocol two uses.

**Comment by jbonnet** [comment]
hey, @mzacharioudakis (CC @trakadasp, @felipevicens ):
I've enriched the `.../api/v2/available-services` end-point; now you can get something like
```javascript
[
    {
        "alive_since": "2017-08-21 11:01:29 UTC", 
        "name": "api", 
        "seconds": 0.497789
    }, 
    {
        "alive_since": "2017-08-21 10:12:02 UTC", 
        "name": "packages", 
        "seconds": 2968.06365
    }, 
    {
        "alive_since": null, 
        "name": "services", 
        "seconds": null
    },...
```
(the `null` is when a micro-service doesn't support -- yet -- the `/began_at` end-point).

I guess that now the `dashboard` can be implemented (services with nulled `alive_since` can be represented as yellow or red)? Feel free to suggest other names, features, etc.

**Comment by jbonnet** [comment]
Closed by #822
